xarray_sql/dataframe_protocol/accessor.py

```diff
@@ -0,0 +1,32 @@
+"""Dataset accessor for the DataFrame interchange protocol."""
+
+from __future__ import annotations
+
+import typing as t
+
+import xarray as xr
+
+from . import core
+
+
+AccessorReturn = t.TypeVar("AccessorReturn")
+
+
+@xr.register_dataset_accessor("df_protocol")
+class DataFrameProtocolAccessor:
+  """Access DataFrame interchange utilities from an xarray Dataset."""
+
+  def __init__(self, dataset: xr.Dataset) -> None:
+    self._dataset = dataset
+
+  def __dataframe__(self, allow_copy: bool = False):
+    """Expose the dataset via the DataFrame interchange protocol."""
+    return core.dataset_to_protocol(self._dataset, allow_copy=allow_copy)
+
+
+def register_dataset_dataframe_accessor() -> None:
```

Comment:
🐑 (the sheep emoji means a sheepish, ie. optional, suggestion). Any ideas for a more succinct name for this method? Further, how will this be used?

Maybe `register_dataframe_interchange()`? Users will have some initial context given the full import path to this method (to imply this applies to xarray datasets).

Reply:
That sounds good to me!